### PR TITLE
updates app-spec to support jsonString

### PR DESCRIPTION
**Pull request by willemolding**
- jsonString replaces serde_json::Value as the json return type from zome functions
- Also changes how API calls are made
- Updates tests to eliminate hard-coded hashes

### CODE REVIEW

**Comment by willemolding** [comment]
Depends on holochain/holochain-rust#490

File: test/test.js [hunk 1]
```diff
@@ -91,43 +68,33 @@ test('my_posts', (t) => {
...
-    post_hash,
-    "QmdJHaznj5rAtMV5nXLK87tdCBoc2NJRtQW4r3w7LZ6HSg"
-  )
-
-  const params_get = JSON.stringify({post_hash})
+  const params_get = JSON.stringify({post_address})
   const result = app.call("blog", "main", "get_post", params_get)
 
   const entry = JSON.parse(result)
+  console.log(entry)
```
Comment (thedavidmeister) [inline test/test.js]:
```suggestion
```

File: zomes/blog/code/src/lib.rs [hunk 2]
```diff
@@ -25,25 +29,25 @@ define_zome! {
...
+                outputs: |result: JsonString|,
                 handler: blog::handle_hash_post
             }
 
-            get_an_address: {
-                inputs: |post_hash: HashString|,
-                outputs: |post: serde_json::Value|,
-                handler: blog::handle_get_an_address
-            }
+            // get_an_address: {
```
Comment (thedavidmeister) [inline zomes/blog/code/src/lib.rs]: cruft?

File: zomes/blog/code/src/blog.rs [hunk 3]
```diff
@@ -84,52 +76,54 @@ pub fn handle_create_post(content: String, in_reply_to: HashString) -> serde_jso
                     let _ = hdk::link_entries(&in_reply_to, &post_hash, "comments");
                 }
             }
-            json!({"hash": post_hash})
+            json!({"address": post_hash}).into()
```
Comment (thedavidmeister) [inline zomes/blog/code/src/blog.rs]: this seems weird to me, manually using `json!` instead of making a struct that derives `DefaultJson` and using that

**Review by thedavidmeister — CHANGES_REQUESTED** [review]

**Review by zippy — APPROVED** [review]
This all looks great (though I agree with @thedavidmeister change requests just as cleanup)  I'd like to see the updates made so we can merge this and get app-spec passing again...

File: zomes/blog/code/src/blog.rs (hunk 3 above)
Comment (willemolding) [reply zomes/blog/code/src/blog.rs]: Sure. I managed to get rid of all uses of the json! macros by adding two extra structs

File: zomes/blog/code/src/lib.rs (hunk 2 above)
Comment (willemolding) [reply zomes/blog/code/src/lib.rs]: removed

**Comment by willemolding** [comment]
Ok @thedavidmeister  changes made as requested

**Comment by willemolding** [comment]
Do I need to do anything fancy to make the docker CI work properly? I can't seem to figure out why this passes locally but not on the CI.

**Comment by Connoropolous** [comment]
@willemolding do you run tests using docker locally? Because that is how you can get tests passing that should pass on travis

**Comment by Connoropolous** [comment]
You run `. docker/run-ci`

**Comment by zippy** [comment]
The error here is with the version QT that's baked into the docker image.  I've updated that, and merged your changes into a new branch with a [PR here](https://github.com/holochain/app-spec-rust/pull/47)   but note that it's still failing for a different problem over in holochain-cm that I have a [different pr for] (https://github.com/holochain/holochain-cmd/pull/91)

**Pull request merged**